modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/FineGrainedControlClassLoader.java

```diff
@@ -48,7 +50,7 @@ public class FineGrainedControlClassLoader extends URLClassLoader
   private final boolean verboseLogging;
 
   public FineGrainedControlClassLoader(URL[] urls, ClassLoader parent, ClassLoaderLookupPolicy lookupPolicy) {
-    super(urls, parent);
+    super(urls, parent, getCachingURLStreamHandlerFactory());
```

Comment:
This would return null folr Java 11+, not ideal but with least changes as we have many child classes, also noth java 8 and java 11 implmentations pass nulll to URLClassLoader constructor when not specified